### PR TITLE
[Behats] Cart summary page - test attributes added

### PR DESCRIPTION
**Pull request by AdamKasp**
| Q               | A
| --------------- | -----
| Branch?         | master
| Bug fix?        | no
New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| License         | MIT

### CODE REVIEW

File: src/Sylius/Behat/Page/Shop/Cart/SummaryPage.php [hunk 1]
```diff
@@ -214,18 +166,12 @@ public function hasItemWithOptionValue($productName, $optionName, $optionValue)
         return $optionValue === $optionValueElement->getText();
     }
```
Comment (Tomanhez) [inline src/Sylius/Behat/Page/Shop/Cart/SummaryPage.php]: I think that you should replace UI classes in this line too : `$selector = sprintf('.sylius-product-options > .item[data-sylius-option-name="%s"]', $optionName);`

File: src/Sylius/Behat/Page/Shop/Cart/SummaryPage.php [hunk 2]
```diff
@@ -214,18 +166,12 @@ public function hasItemWithOptionValue($productName, $optionName, $optionValue)
...
 
-    /**
-     * {@inheritdoc}
-     */
-    public function hasItemWithCode($code)
+    public function hasItemWithCode(string $code): bool
     {
         return $this->hasItemWith($code, '.sylius-product-variant-code');
```
Comment (GSadee) [inline src/Sylius/Behat/Page/Shop/Cart/SummaryPage.php]: I guess you should use test attributes also here

File: src/Sylius/Behat/Page/Shop/Cart/SummaryPage.php [hunk 3]
```diff
@@ -236,28 +182,19 @@ public function hasProductOutOfStockValidationMessage(ProductInterface $product)
...
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function isEmpty()
+    public function isEmpty(): bool
     {
-        return false !== strpos($this->getDocument()->find('css', '.message')->getText(), 'Your cart is empty');
+        return false !== strpos($this->getDocument()->find('css', '[data-test-flash-message]')->getText(), 'Your cart is empty');
```
Comment (GSadee) [inline src/Sylius/Behat/Page/Shop/Cart/SummaryPage.php]: Use `getElement`

File: src/Sylius/Behat/Page/Shop/Cart/SummaryPage.php [hunk 4]
```diff
@@ -134,77 +110,53 @@ public function getItemImage(int $itemNumber): string
...
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function changeQuantity($productName, $quantity)
+    public function changeQuantity(string $productName, string $quantity): void
     {
         $itemElement = $this->getElement('product_row', ['%name%' => $productName]);
         $itemElement->find('css', 'input[type=number]')->setValue($quantity);
```
Comment (GSadee) [inline src/Sylius/Behat/Page/Shop/Cart/SummaryPage.php]: Couldn't be as one element?

File: src/Sylius/Bundle/ShopBundle/Resources/views/Cart/_widget.html.twig [hunk 5]
```diff
@@ -4,7 +4,7 @@
     {{ sonata_block_render_event('sylius.shop.partial.cart.summary.before_widget_content', {'cart': cart}) }}
 
     <i class="cart icon"></i>
-    <span id="sylius-cart-total">
+    <span id="sylius-cart-total" {{ sylius_test_html_attribute("cart-total") }}>
```
Comment (GSadee) [inline src/Sylius/Bundle/ShopBundle/Resources/views/Cart/_widget.html.twig]:
```suggestion
    <span id="sylius-cart-total" {{ sylius_test_html_attribute('cart-total') }}>
```

File: src/Sylius/Behat/Page/Shop/Cart/SummaryPage.php [hunk 6]
```diff
@@ -134,77 +110,51 @@ public function getItemImage(int $itemNumber): string
...
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function removeProduct($productName)
+    public function removeProduct(string $productName): void
     {
         $itemElement = $this->getElement('product_row', ['%name%' => $productName]);
         $itemElement->find('css', 'button.sylius-cart-remove-button')->press();
```
Comment (GSadee) [inline src/Sylius/Behat/Page/Shop/Cart/SummaryPage.php]: Could be in one element

**Comment by GSadee** [comment]
Thank you, Adam! :tada:

**Pull request merged**